core/src/main/java/org/apache/iceberg/BaseFilesTable.java

```diff
@@ -56,72 +56,73 @@ public Schema schema() {
     }
   }
 
-  abstract static class BaseFilesTableScan extends BaseMetadataTableScan {
+  private static CloseableIterable<FileScanTask> planFiles(Table table, CloseableIterable<ManifestFile> manifests,
```

Comment:
Moved the common planning logic to a static method to share in both scans.

Reply:
I also removed the comment about column pruning as I think it is no longer accurate. We do support column pruning for metadata tables.